src/mlang/mpp_ir/mpp_ir_to_bir.ml

```diff
@@ -470,26 +478,33 @@ let create_combined_program (m_program : Mir_interface.full_program)
     Bir.program =
   try
     let mpp_program = List.rev mpp_program in
-    let decl_to_extract =
-      try
-        List.find
-          (fun decl -> decl.Mpp_ir.name = mpp_function_to_extract)
-          mpp_program
-      with Not_found ->
-        Errors.raise_error
-          (Format.asprintf "M++ function %s not found in M++ file!"
-             mpp_function_to_extract)
-    in
-    let ctx, statements =
-      translate_mpp_function mpp_program m_program decl_to_extract []
-        emtpy_translation_ctx
+    let ctx, mpp_functions =
+      List.fold_left
+        (fun (ctx, function_map) mpp_func ->
+          let ctx, statements =
+            translate_mpp_function mpp_program m_program mpp_func [] ctx
+          in
+          (ctx, Bir.FunctionMap.add mpp_func.name statements function_map))
+        (emtpy_translation_ctx, Bir.FunctionMap.empty)
+        mpp_program
     in
+    if not (Bir.FunctionMap.mem mpp_function_to_extract mpp_functions) then
+      Errors.raise_error
+        (Format.asprintf "M++ function %s not found in M++ file!"
+           mpp_function_to_extract);
     let conds_verif = generate_verif_conds m_program.program.program_conds in
+    let mpp_main_function =
+      Bir.FunctionMap.find mpp_function_to_extract mpp_functions
+    in
+    let mpp_functions =
+      Bir.FunctionMap.add mpp_function_to_extract
+        (mpp_main_function @ conds_verif)
```

Comment:
Again, injecting verifications in a user defined function might become unacceptable later, fine for now as it keeps the way it worked to this point. Please comment.